File: image-scripts/convertImageToMatrix.js
const fs = require('fs');

function loadMatrixFromFile(filePath) {
  const data = fs.readFileSync(filePath, 'utf-8');
  return data
    .trim()
    .split('\n')
    .map((line) => line.split('').map(Number));
}

function saveMatrixToJson(matrix, jsonPath) {
  fs.writeFileSync(jsonPath, JSON.stringify(matrix, null, 2));
  console.log(`Matrice enregistrée dans ${jsonPath}`);
}

const matrix = loadMatrixFromFile('image-scripts/output/naolibMatrix.txt');
saveMatrixToJson(matrix, 'image-scripts/output/naolibMatrix.json');

File: image-scripts/convert_image_to_matrix.py
from PIL import Image

def image_to_matrix(image_path, width=64, height=32):
    img = Image.open(image_path)
    img = img.resize((width, height))
    img = img.convert('RGB')
    pixels = list(img.getdata())
    matrix = []
    for y in range(height):
        row = []
        for x in range(width):
            r, g, b = pixels[y * width + x]
            row.append(1 if g > 128 and r < 128 and b < 128 else 0)
        matrix.append(row)
    return matrix

def save_matrix_to_file(matrix, file_path):
    with open(file_path, 'w') as f:
        for row in matrix:
            f.write(''.join(map(str, row)) + '\n')

matrix = image_to_matrix('image-scripts/input/naolib.png')
save_matrix_to_file(matrix, 'image-scripts/output/naolibMatrix.txt')